pkg/freeze/containerd/pause_test.go

```diff
@@ -0,0 +1,124 @@
+package containerd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"google.golang.org/grpc"
+	cri "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"knative.dev/container-freezer/pkg/daemon"
+)
+
+type FakeContainerdCRI struct {
+	paused     []string
+	resumed    []string
+	containers []*cri.Container
+	results    []string
+	method     string
+}
+
+// Container creates a CRI container with the given container ID and name
+func Container(id, name string) *cri.Container {
+	return &cri.Container{
+		Id: id,
+		Metadata: &cri.ContainerMetadata{
+			Name: name,
+		},
+	}
+}
+
+func (c *FakeContainerdCRI) List(ctx context.Context, conn *grpc.ClientConn, podUID string) (*cri.ListContainersResponse, error) {
+	return &cri.ListContainersResponse{Containers: c.containers}, nil
+}
+
+func (c *FakeContainerdCRI) Pause(ctx context.Context, ctrd *containerd.Client, container string) error {
+	c.paused = append(c.paused, container)
+	if c.method != "freeze" {
+		return fmt.Errorf("wrong method, expected: %s, got: %s", "freeze", c.method)
+	}
+	return nil
+}
+
+func (c *FakeContainerdCRI) Resume(ctx context.Context, ctrd *containerd.Client, container string) error {
+	c.resumed = append(c.resumed, container)
```

Comment:
Presumably there should be something asserting on this in the actual test (unless I missed it)

Reply:
The Thaw/Freeze methods will return an error if all containers aren't frozen (see [here](https://github.com/knative-sandbox/container-freezer/blob/9262280e0c936790f618b4600aa4a9bb3f9cf089/pkg/freeze/containerd/pause.go#L74-L76)), so I _think_ that should cover us.

Reply:
Alternatively, we could have the Thaw method return the list of thawed containers, which now that I think of it, would probably make testing easier...

Reply:
Implemented the alternative, let me know if it looks ok or if I should scrap it.